File: sdks/python/sealstack/namespaces/schemas.py
"""Schemas namespace (read-only)."""

from typing import Any

from .._http import HttpClient


class SchemasNamespace:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def list(self) -> list[Any]:
        out = await self._http.request("GET", "/v1/schemas")
        return out["schemas"]

    async def get(self, qualified: str) -> Any:
        return await self._http.request("GET", f"/v1/schemas/{qualified}")

File: sdks/typescript/src/namespaces/schemas.ts
import type { HttpClient } from "../http.js";

export class SchemasNamespace {
  constructor(private http: HttpClient) {}

  async list(): Promise<unknown[]> {
    const data = await this.http.request<{ schemas: unknown[] }>({
      method: "GET", path: "/v1/schemas",
    });
    return data.schemas;
  }

  async get(qualified: string): Promise<unknown> {
    return this.http.request({
      method: "GET", path: `/v1/schemas/${encodeURIComponent(qualified)}`,
    });
  }
}
